compiler/packages/babel-plugin-react-compiler/src/TypeInference/InferTypes.ts

```diff
@@ -250,6 +250,7 @@ function* generateInstructionTypes(
     }
 
     case 'CallExpression': {
+      const returnType = makeType();
```

Comment:
Hmm I don't quite understand -- what does creating an intermediate `returnType` do here?

Reply:
It doesn't really matter in practice, but this ensures we have a place to put the inferred return type of the function (eg from its signature) separately from what we infer about how it's lvalue is used. For example in `foo() && ...`, we would inver the lvalue of the `foo()` call is a primitive bc of the logical expression, but that means the actual call expression's return type information is lost. This way we can still track the actual function return type.